command.go

```diff
@@ -0,0 +1,121 @@
+package githubactions
+
+import (
+	"encoding/json"
+	"fmt"
+	"sort"
+	"strings"
+)
+
+// CommandProperties is a named "map[string]interface{}" type to hold key-value
+// pairs passed to an actions command.
+type CommandProperties map[string]interface{}
+
+// String encodes the CommandProperties to a string as comma separated
+// 'key=value' pairs. The pairs are joined in a chronological order.
+func (props *CommandProperties) String() string {
+	l := make([]string, 0, len(*props))
+	for k, v := range *props {
+		l = append(l, fmt.Sprintf("%s=%s", k, escapeProperty(v)))
+	}
+
+	sort.Strings(l)
+	return strings.Join(l, ",")
+}
+
+// Command can be issued by a GitHub action by writing to `stdout` with
+// following format.
+//
+// ::name key=value,key=value::message
+//
+//  Examples:
+//    ::warning::This is the message
+//    ::set-env name=MY_VAR::some value
+type Command struct {
+	Name       string
+	Message    interface{}
+	Properties CommandProperties
+}
+
+// String encodes the Command to a string in the following format:
+//
+// ::name key=value,key=value::message
+func (cmd *Command) String() string {
+	// https://github.com/actions/toolkit/blob/9ad01e4fd30025e8858650d38e95cfe9193a3222/packages/core/src/command.ts#L43-L45
+	if cmd.Name == "" {
+		cmd.Name = "missing.command"
+	}
+
+	const cmdSeparator = "::"
+	var builder strings.Builder
+	builder.WriteString(cmdSeparator)
+	builder.WriteString(cmd.Name)
+	if len(cmd.Properties) > 0 {
+		builder.WriteString(" ")
+		builder.WriteString(cmd.Properties.String())
+	}
+
+	builder.WriteString(cmdSeparator)
+	builder.WriteString(escapeData(cmd.Message))
+	return builder.String()
+}
+
+// toCommandValue sanitizes an input into a string so it can be passed with a
+// Command safely.
+//
+// The equivalent toolkit function can be found here:
+//
+// https://github.com/actions/toolkit/blob/9ad01e4fd30025e8858650d38e95cfe9193a3222/packages/core/src/command.ts#L83-L90
+func toCommandValue(i interface{}) string {
+	switch v := i.(type) {
+	case nil:
+		return ""
+	case string:
+		return v
+	case fmt.Stringer:
+		return v.String()
+	default:
+		data, err := json.Marshal(v)
+		if err != nil {
+			panic(err)
```

Comment:
I'm a bit -1 on panicing. This should return an error and string and propagate up the stack.

Reply:
On second thought, propagating the error upwards will cause changes in signatures for all `Action.*` functions. Should I go ahead and make the changes?

Reply:
Hmmm... Looking closer, I think `toCommandValue` should always accept a string? I'm not seeing where it could accept anything else?

Reply:
In general cases, yes, but the TypeScript counterpart accepts generic input and JSON encodes non-string values. Maybe we're better off with strings!

Reply:
Digging deeper into `json.Marshal()`, it should only produce an error on complex values (functions, channels) and cyclic data types. It wouldn't raise an error otherwise. https://golang.org/pkg/encoding/json/#Marshal

I think both of the scenarios are justified for a panic.

Reply:
I think I'd prefer to keep it as strings for now and avoid the panic. I haven't really seen a use case where the value isn't a string.

Reply:
Okay! Making the changes.

Reply:
There's one more panic though: https://github.com/sethvargo/go-githubactions/pull/10/files#diff-f52cd89f8b009086d7ca39479e79ed3ad8643af87403f44103f9a00af45b426fR217

Reply:
yea I left a comment - I'm not sure how I feel about converting the slice to a map. I'd rather convert the map to a slice.